v2/pkg/genruntime/extensions/successful_resource_modifier.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package extensions
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+// SuccessfulResourceModifier can be implemented to customize the resource upon successful creation
+type SuccessfulResourceModifier interface {
+	// Success modifies the resource based on a successful creation
+	Success(obj genruntime.ARMMetaObject) error
+}
```

Comment:
What's the scenario where this extension point is needed?

Reply:
Because `subscriptionAliases` don't have the right url to assign child resources (like RoleAssignment) to, I've got in another branch the addition of a concept called `ChildResourceIDOverride`, which is an annotation that can be added to an object. If that annotation is added to the object, when child resources are attached to that object, instead of using the normal ID/Type of the object they just use the URL in the annotation. This hook is so that I can manually add that annotation to the `subscriptionAlias` type, which then causes it to be able to correctly have child resources created off of the subscription (rather than off of the alias).

Basically it's a way to combine two separate resources into 1. It's somewhat hacky so we shouldn't really use it regularly, but it saves us from having to implement our own reconciler for subscription, it can just use the normal reconciler.

Reply:
Good use of an extension point.